pkg/autoscaler/autoscaler.go

```diff
@@ -199,24 +203,39 @@ func (agg *perPodAggregation) podWeight(now time.Time) float64 {
 // Autoscaler stores current state of an instance of an autoscaler
 type Autoscaler struct {
 	*DynamicConfig
-	key          string
-	target       float64
-	stats        map[statKey]Stat
-	statsMutex   sync.Mutex
-	panicking    bool
-	panicTime    *time.Time
-	maxPanicPods float64
-	reporter     StatsReporter
-	targetMutex  sync.RWMutex
+	key             string
+	namespace       string
+	revisionService string
+	endpointsLister corev1listers.EndpointsLister
+	target          float64
+	stats           map[statKey]Stat
+	statsMutex      sync.Mutex
+	readyPods       map[time.Time]float64
+	readyPodsMutex  sync.RWMutex
+	panicking       bool
+	panicTime       *time.Time
+	maxPanicPods    float64
+	reporter        StatsReporter
+	targetMutex     sync.RWMutex
```

Comment:
you have 3 mutexes in this stucture.
Please comment which mutex guards which element.
E.g.
```
// m1 guards the elements in the block below.
m1 sync.Mutex
a int
b map[]
...

// m2 guards ...
m2 sync.Mutex
```